connection.go

```diff
@@ -152,6 +152,34 @@ func (c *Connection) Connect() error {
 	return nil
 }
 
+// Reconnect establishes a new connection
+func (c *Connection) Reconnect() error {
+	// Should only be called on non-active channels, this adds additional protections
+	err := c.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close connection")
+	}
+
+	var conn net.Conn
+	d := &net.Dialer{Timeout: c.Opts.ConnectTimeout}
+
+	if c.Opts.TLSConfig != nil {
+		conn, err = tls.DialWithDialer(d, "tcp", c.Addr, c.Opts.TLSConfig)
+	} else {
+		conn, err = d.Dial("tcp", c.Addr)
+	}
```

Comment:
IMO we should re-use this code from the initial connect call.

Reply:
@adamdecaf agreed, this was just a quick and dirty "is this ok, here's the other thought" - I'll definitely clean up and ad unit tests for non-proof-of-concept

Reply:
Can you try `go get github.com/moov-io/iso8583-connection@CAR-814-add-reconnect-option` in your project then? We can get it working in your app before merging to this repository.

Reply:
I already did, tested it and worked as intended

Reply:
Here is PR, pushed it up: https://github.com/moovfinancial/disco-gateway/pull/62